azure/azure.go

```diff
@@ -114,6 +251,88 @@ func (a *azureOps) InstanceID() string {
 	return a.instance
 }
 
+func (a *azureOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+
+	instInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name:   a.instance,
+			ID:     a.instance,
+			Zone:   "",
+			Region: "",
+		},
+	}
+	return instInfo, nil
+}
+
+func (a *azureOps) InspectInstanceGroupForInstance(instanceID string) (*cloudops.InstanceGroupInfo, error) {
+
+	ctx := context.Background()
+	agentPool, err := a.agentPoolsClient.Get(ctx, a.resourceGroupName, a.managedClusterName, a.agentPoolName)
+	if err != nil {
+		logrus.Errorf("Failed to get node pool details for [%s] of cluster [%v] in [%v] resource group."+
+			"Error: [%v] ", a.agentPoolName, a.managedClusterName, a.resourceGroupName, err)
+		return nil, err
+	}
+
+	// TODO: Currently multi-availability feature is in preview and
```

Comment:
@piyush-nimbalkar should review this. Odd for azure API to not support zones and regions.

Reply:
@harsh-px / @piyush-nimbalkar , when I try to get agentPool details by using `a.agentPoolsClient.Get()` API call, I get bellow response.

For Availability Sets,
<img width="1164" alt="Screenshot 2020-01-07 at 2 08 14 PM" src="https://user-images.githubusercontent.com/4953411/71880946-5ff48e80-3157-11ea-910d-046e32fbe9b1.png">
```
AgentPool details: [containerservice.AgentPool{Response:autorest.Response{Response:(*http.Response)(0xc00027ca20)}, ManagedClusterAgentPoolProfileProperties:(*containerservice.ManagedClusterAgentPoolProfileProperties)(0xc000128100), ID:(*string)(0xc0004d5550), Name:(*string)(0xc0004d5560), Type:(*string)(0xc0004d51e0)}]

Availability Zones: [(*[]string)(nil)]

agentPool.ManagedClusterAgentPoolProfileProperties: [&containerservice.ManagedClusterAgentPoolProfileProperties{Count:(*int32)(0xc0004c5568), VMSize:"Standard_DS3_v2", OsDiskSizeGB:(*int32)(0xc0004c556c), VnetSubnetID:(*string)(nil), MaxPods:(*int32)(0xc0004c55a4), OsType:"Linux", MaxCount:(*int32)(nil), MinCount:(*int32)(nil), EnableAutoScaling:(*bool)(nil), Type:"AvailabilitySet", OrchestratorVersion:(*string)(0xc0004d5540), ProvisioningState:(*string)(0xc0004d5530), AvailabilityZones:(*[]string)(nil)}]

agentPool.ManagedClusterAgentPoolProfileProperties.AvailabilityZones: [(*[]string)(nil)]

agentPool.EnableAutoScaling: [(*bool)(nil)]

agentPool.MaxCount: [<nil>], agentPool.MinCount: [<nil>]
```

For VMSS scale set,
<img width="1278" alt="Screenshot 2020-01-07 at 2 12 12 PM" src="https://user-images.githubusercontent.com/4953411/71881232-ead58900-3157-11ea-9627-d1d10a903201.png">

```
AgentPool details: [containerservice.AgentPool{Response:autorest.Response{Response:(*http.Response)(0xc0001f02d0)}, ManagedClusterAgentPoolProfileProperties:(*containerservice.ManagedClusterAgentPoolProfileProperties)(0xc0001fa380), ID:(*string)(0xc0004dd660), Name:(*string)(0xc0004dd670), Type:(*string)(0xc0004dd390)}]

Availability Zones: [(*[]string)(nil)]
agentPool.ManagedClusterAgentPoolProfileProperties: [&containerservice.ManagedClusterAgentPoolProfileProperties{Count:(*int32)(0xc00003c088), VMSize:"Standard_DS3_v2", OsDiskSizeGB:(*int32)(0xc00003c08c), VnetSubnetID:(*string)(nil), MaxPods:(*int32)(0xc00003c0c4), OsType:"Linux", MaxCount:(*int32)(nil), MinCount:(*int32)(nil), EnableAutoScaling:(*bool)(nil), Type:"VirtualMachineScaleSets", OrchestratorVersion:(*string)(0xc0004dd650), ProvisioningState:(*string)(0xc0004dd640), AvailabilityZones:(*[]string)(nil)}]

agentPool.ManagedClusterAgentPoolProfileProperties.AvailabilityZones: [(*[]string)(nil)]

agentPool.EnableAutoScaling: [(*bool)(nil)]

agentPool.MaxCount: [<nil>], agentPool.MinCount: [<nil>]

```

Reply:
I just did this from azure cli (with debug to see what API it calls), which calls the same API as you are and it has zone information - 
```
az aks nodepool show --cluster-name piyush-test-mz -g piyush-test-mz --name nodepool1 --debug
```
```
GET /subscriptions/XXX/resourceGroups/piyush-test-mz/providers/Microsoft.ContainerService/managedClusters/piyush-test-mz/agentPools/nodepool1?api-version=2019-08-01

{
  "agentPoolType": "VirtualMachineScaleSets",
  "availabilityZones": [
    "1",
    "2",
    "3"
  ],
  "count": 0,
  "enableAutoScaling": true,
  "enableNodePublicIp": null,
  "id": "/subscriptions/XXXX/resourcegroups/piyush-test-mz/providers/Microsoft.ContainerService/managedClusters/piyush-test-mz/agentPools/nodepool1",
  "maxCount": 6,
  "maxPods": 110,
  "minCount": 3,
  "name": "nodepool1",
  "nodeTaints": null,
  "orchestratorVersion": "1.13.10",
  "osDiskSizeGb": 100,
  "osType": "Linux",
  "provisioningState": "Succeeded",
  "resourceGroup": "piyush-test-mz",
  "scaleSetEvictionPolicy": null,
  "scaleSetPriority": null,
  "type": "Microsoft.ContainerService/managedClusters/agentPools",
  "vmSize": "Standard_DS2_v2",
  "vnetSubnetId": null
}
```
The cluster I tried was a multi-zone aks cluster.
When an AKS cluster does not in AZs it is running in Availability Sets (AS) which is a different construct than AZ. 
A VM can run either in an AZ or AS. So we should populate the values if present else may be set an empty array instead of dummy value.
@harsh-px @vinayakshnd - How is the Zones field used in cloudops or it's user?

Reply:
@piyush-nimbalkar currently zone field is not used in any other API other than in test here:
https://github.com/libopenstorage/cloudops/blob/master/test/cloudops.go#L115

For the above test, if empty/nil availability zone array comes, I still need `len(groupInfo.Zones)` to be 1 to calculate the expected size of cluster.